design/apps.go

```diff
@@ -0,0 +1,278 @@
+	a.Security("jwt")
+
+	a.Action("showSpace", func() {
+		a.Routing(
+			a.GET("/spaces/:spaceID"),
+		)
+		a.Description("list applications in a space")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+		})
+		a.Response(d.OK, simpleSpaceSingle)
+		a.Response(d.Unauthorized, JSONAPIErrors)
+		a.Response(d.InternalServerError, JSONAPIErrors)
+		a.Response(d.NotFound, JSONAPIErrors)
+	})
+
+	a.Action("showSpaceApp", func() {
+		a.Routing(
+			a.GET("/spaces/:spaceID/applications/:appName"),
+		)
+		a.Description("list application")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+			a.Param("appName", d.String, "Name of the application")
+		})
+		a.Response(d.OK, simpleAppSingle)
+		a.Response(d.Unauthorized, JSONAPIErrors)
+		a.Response(d.InternalServerError, JSONAPIErrors)
+		a.Response(d.NotFound, JSONAPIErrors)
+	})
+
+	a.Action("showSpaceAppDeployment", func() {
+		a.Routing(
+			a.GET("/spaces/:spaceID/applications/:appName/deployments/:deployName"),
+		)
+		a.Description("list pipe element")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+			a.Param("appName", d.String, "Name of the application")
+			a.Param("deployName", d.String, "Name of the pipe deployment")
+		})
+		a.Response(d.OK, simpleDeploymentSingle)
+		a.Response(d.Unauthorized, JSONAPIErrors)
+		a.Response(d.InternalServerError, JSONAPIErrors)
+		a.Response(d.NotFound, JSONAPIErrors)
+	})
+
+	a.Action("showDeploymentStats", func() {
+		a.Routing(
+			a.GET("/spaces/:spaceID/applications/:appName/deployments/:deployName/stats"),
+		)
+		a.Description("list deployment statistics")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+			a.Param("appName", d.String, "Name of the application")
+			a.Param("deployName", d.String, "Name of the deployment")
+		})
+		a.Response(d.OK, simpleDeploymentSingle)
+		a.Response(d.Unauthorized, JSONAPIErrors)
+		a.Response(d.InternalServerError, JSONAPIErrors)
+		a.Response(d.NotFound, JSONAPIErrors)
+	})
+
+	a.Action("showDeploymentStatSeries", func() {
+		a.Routing(
+			a.GET("/spaces/:spaceID/applications/:appName/deployments/:deployName/statseries"),
+		)
+		a.Description("list deployment statistics")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+			a.Param("appName", d.String, "Name of the application")
+			a.Param("deployName", d.String, "Name of the deployment")
+			a.Param("start", d.Number, "start time in millis")
+			a.Param("end", d.Number, "end time in millis")
+			a.Param("limit", d.Integer, "maximum number of data points to return")
+		})
+		a.Response(d.OK, simpleDeploymentStatSeriesSingle)
+		a.Response(d.Unauthorized, JSONAPIErrors)
+		a.Response(d.InternalServerError, JSONAPIErrors)
+		a.Response(d.NotFound, JSONAPIErrors)
+	})
+
+	a.Action("setDeployment", func() {
+		a.Routing(
+			a.PUT("/spaces/:spaceID/applications/:appName/deployments/:deployName/control"),
```

Comment:
should it be a plural for `control` ?

Reply:
I don't know.  It's not asking to return an object.  I can change it if it seems more consistent.
I was tempted to do
  /spaces/:spaceID/applications/:appName/deployments/:deployName?podcount=99
what do you think? (If I leave it as is for now the UI doesn't need a PR)

Reply:
could you elaborate on the  meaning/usage of the `podcount=99` query param that you're suggesting ?

Reply:
'podcount' is sets the number of active pods (using the up down arrows in the UI).  99 was just an example placeholder.

Reply:
no, I think that the parameters should be part of the request body, not its URL if you use the `PUT` or `POST` verbs. I'm just not comfortable with the `control` verb/action/term in the URL, but you could change that while dealing with the JSON-API compliance in a subsequent PR, I believe.